File: src/main/java/com/github/jnhyperion/hyperrobotframeworkplugin/ide/PostStartupActivity.java
package com.github.jnhyperion.hyperrobotframeworkplugin.ide;

import com.github.jnhyperion.hyperrobotframeworkplugin.MyLogger;
import com.intellij.openapi.Disposable;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.startup.ProjectActivity;
import kotlin.Unit;
import kotlin.coroutines.Continuation;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class PostStartupActivity implements Disposable, ProjectActivity {

    @Override
    public void dispose() {
        MyLogger.logger.debug("dispose");
    }

    @Override
    public @Nullable Object execute(@NotNull Project project, @NotNull Continuation<? super Unit> continuation) {
        MyLogger.logger.debug("runActivity: " + project.getName());
        RobotListenerMgr.getInstance().initializeListeners(project);
        return null;
    }
}

File: src/main/kotlin/dev/xeonkryptos/xeonrobotframeworkplugin/PostStartupActivity.kt
package dev.xeonkryptos.xeonrobotframeworkplugin

import com.intellij.openapi.project.Project
import com.intellij.openapi.roots.ProjectRootManager
import com.intellij.openapi.startup.ProjectActivity
import dev.xeonkryptos.xeonrobotframeworkplugin.util.RobotVersionUpdaterService
import kotlin.time.Duration.Companion.minutes

class PostStartupActivity : ProjectActivity {

    override suspend fun execute(project: Project) {
        RobotListenerMgr.getInstance(project).initializeListeners()
        ProjectRootManager.getInstance(project).projectSdk

        RobotVersionUpdaterService.getInstance(project).startVersionUpdater(1.minutes)
    }
}
